R/shuffle_ard.R

```diff
@@ -223,8 +224,119 @@ shuffle_ard <- function(x, trim = TRUE) {
   }
 }
 
+#' Rename ARD Columns
+#'
+#' This function combines a pair of `group`/`group_level` or `variable`/`variable_level` columns into a
+#' single column. The `group_level` or `variable_level` column is renamed according to the value of
+#' the `group` or `variable` column, respectively.
+#'
+#' @param x (`data.frame`)\cr
+#'   a data frame
+#' @param col ([`tidy-select`][dplyr::dplyr_tidy_select])\cr
+#'  Name of column containing the variable names
+#' @param col_lev ([`tidy-select`][dplyr::dplyr_tidy_select])\cr
+#'  Name of column containing the variable levels. If `NULL` (default), this will be assumed to match the `col` argument with `_level` appended to the end (e.g. if `col` = "group1", `col_lev` will be assumed to be "group1_level")
+#'
+#' @return data frame
+#' @export
+#'
+#' @examples
+#' data <- data.frame(group1 = "A", group1_level = "B", variable = "C", variable_level = "D")
+#'
+#' cards:::rename_ard_columns(data, "group1")
+#' cards:::rename_ard_columns(data, "variable")
+rename_ard_columns <- function(x, col, col_lev = NULL) {
```

Comment:
For the API, can we simplify it a bit to be 

```r
rename_ard_columns <- function(x, columns = c(all_ard_groups(), all_ard_variables()))
```

With each column name/level pair, we can loop over this function you have here.

Reply:
I like that idea (and prefer it)! I just was trying to accommodate the situation where someone would want to do this following a `shuffle_ard()` call which would be `rename_ard_columns(data, col = "variable", col_lev = "label")`. But maybe this tradeoff isn't worth it. Can we discuss?

Reply:
We will have this API with this default

```r
rename_ard_columns <- function(x, columns = c(all_ard_groups(), all_ard_variables()))
```

If a user need to rename after a shuffle, then they would specify columns `c("variable", "label")` and we'll handle that case internally

Reply:
@ddsjoberg Updated! I missed that we already had an internal `.rename_ard_columns` so I decided to rename this to `shift_ard_columns`. Let me know what you think!

Reply:
hey @bzkrouse ! I renamed the internal function so we could keep our original function name. I think it's more clear to the users this way